src/averageAges.js

```diff
@@ -15,11 +15,19 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const agesMans = people.map(person => (person.sex === 'm'
+  && century === undefined)
+  || (person.sex === 'm' && Math.ceil(person.died / 100) === century)
+    ? person.died - person.born
+    : undefined).filter(age => age !== undefined);
```

Comment:
Fix such checkers everywhere
```suggestion
    : undefined).filter(age => age);
```